### PR TITLE
Updated linting

### PR DESCRIPTION
**Pull request by fredbi**
* updated golangci lint rules
* updated github actions
* updated dependencies
* fixed dead links/badges in README
* relinted (tests and docstrings)
* removed unexecuted integration tests

### CODE REVIEW

File: loads_test/spec_circular_test.go [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-
-	"github.com/go-openapi/loads"
-	"github.com/go-openapi/strfmt"
-	"github.com/go-openapi/validate"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-)
-
-func TestPathLoaderIssue(t *testing.T) {
-	swaggerFile := "fixtures/json/resources/pathLoaderIssue.json"
```
Comment (youyuanwu) [inline loads_test/spec_circular_test.go:14]: Why is this removed?

**Review by youyuanwu — APPROVED** [review]

File: loads_test/spec_circular_test.go (hunk 1 above)
Comment (fredbi) [reply loads_test/spec_circular_test.go:14]: @youyuanwu 
Integration tests are not executed by CI, so this is essentially almost dead code.
The maintainance of the dependencies for this separate module is going to be a pain on the long run.
I believe we have better ways with Github actions to run intgration tests without adding cross-dependencies like this. Working on that part.
